duneana/CAFMaker/CAFMaker_module.cc

```diff
@@ -172,302 +199,461 @@ namespace caf {
+    std::string geoName = fGeometry->DetectorName();
 
-    if( !pidinnue.failedToGet() ) {
-      sr.mvanue = pidinnue->pid;
-    }
+    std::cout << "Geo name is: " << geoName << std::endl;
 
-    if( !pidinnumu.failedToGet() ) {
-      sr.mvanumu = pidinnumu->pid;
-    }
+    SRDetectorMeta *detector;
 
-    if( !cvnin.failedToGet() ) {
-      //using i = cvn::Interaction;
-      //if(cvnin->empty() || (*cvnin)[0].fOutput.size() <= i::kNutauOther){
-      if(cvnin->empty()){
-        sr.CVNResultIsAntineutrino = sr.CVNResultNue = sr.CVNResultNumu = sr.CVNResultNutau = sr.CVNResultNC = \
-        sr.CVNResult0Protons = sr.CVNResult1Protons = sr.CVNResult2Protons = sr.CVNResultNProtons = \
-        sr.CVNResult0Pions = sr.CVNResult1Pions = sr.CVNResult2Pions = sr.CVNResultNPions = \
-        sr.CVNResult0Pizeros = sr.CVNResult1Pizeros = sr.CVNResult2Pizeros = sr.CVNResultNPizeros = \
-        sr.CVNResult0Neutrons = sr.CVNResult1Neutrons = sr.CVNResult2Neutrons = sr.CVNResultNNeutrons = -3;
-      }
-      else{
-        //const std::vector<float>& v = (*cvnin)[0].fOutput;
-        //sr.CVNResultNue = v[i::kNueQE] + v[i::kNueRes] + v[i::kNueDIS] + v[i::kNueOther];
-        //sr.CVNResultNumu = v[i::kNumuQE] + v[i::kNumuRes] + v[i::kNumuDIS] + v[i::kNumuOther];
-        //sr.CVNResultNutau = v[i::kNutauQE] + v[i::kNutauRes] + v[i::kNutauDIS] + v[i::kNutauOther]
-
-        sr.CVNResultIsAntineutrino = (*cvnin)[0].GetIsAntineutrinoProbability();
-
-        sr.cvnnue = (*cvnin)[0].GetNueProbability();
-        sr.cvnnumu = (*cvnin)[0].GetNumuProbability();
-        sr.cvnnutau = (*cvnin)[0].GetNutauProbability();
-        sr.cvnnc = (*cvnin)[0].GetNCProbability();
-
-        sr.CVNResult0Protons = (*cvnin)[0].Get0protonsProbability();
-        sr.CVNResult1Protons = (*cvnin)[0].Get1protonsProbability();
-        sr.CVNResult2Protons = (*cvnin)[0].Get2protonsProbability();
-        sr.CVNResultNProtons = (*cvnin)[0].GetNprotonsProbability();
-
-        sr.CVNResult0Pions = (*cvnin)[0].Get0pionsProbability();
-        sr.CVNResult1Pions = (*cvnin)[0].Get1pionsProbability();
-        sr.CVNResult2Pions = (*cvnin)[0].Get2pionsProbability();
-        sr.CVNResultNPions = (*cvnin)[0].GetNpionsProbability();
-
-        sr.CVNResult0Pizeros = (*cvnin)[0].Get0pizerosProbability();
-        sr.CVNResult1Pizeros = (*cvnin)[0].Get1pizerosProbability();
-        sr.CVNResult2Pizeros = (*cvnin)[0].Get2pizerosProbability();
-        sr.CVNResultNPizeros = (*cvnin)[0].GetNpizerosProbability();
-
-        sr.CVNResult0Neutrons = (*cvnin)[0].Get0neutronsProbability();
-        sr.CVNResult1Neutrons = (*cvnin)[0].Get1neutronsProbability();
-        sr.CVNResult2Neutrons = (*cvnin)[0].Get2neutronsProbability();
-        sr.CVNResultNNeutrons = (*cvnin)[0].GetNneutronsProbability();
-      }
+    if(geoName.find("dunevd10kt") != std::string::npos){
+      detector = &(sr.meta.fd_vd);
+      std::cout << "Assuming the FD VD detector" << std::endl;
```

Comment:
What does the detector info get used for?  Does there need to be a distinction between HD and VD?

Reply:
Corresponds to distinct branches in the new CAF format for the meta info. I am not sure of the relevance, I just tried to follow the pattern.